### PR TITLE
fix(docs): render typedoc module index members as compact tables instead of bullet lists

### PR DESCRIPTION
**Pull request by kazupon**
## Background

Under `generateDocsMarkdown(..., { pathStrategy: 'typedoc', renderStyle: 'markdown' })`, each module `index.md` listed its members as a **bullet list** that inlined every symbol's **entire type signature**:

```md
## Functions

- [`cli`](/api-ox/default/functions/cli.md) `function` `cli<G extends GunshiParamsConstraint>(args: string[], entry: Command<G> | CommandRunner<G> | LazyCommand<G>, options?: CliOptions<G>): Promise<string | undefined>` - Run the command.
- ... (one bullet per overload)
```

TypeDoc renders the same index as a compact `Name | Description` table, with the signature living only on the per-symbol page:

```md
## Functions

| Function | Description |
| ------ | ------ |
| [cli](functions/cli.md) | Run the command. |
```

The bullet-with-inlined-signature form makes the index extremely long (a single generic signature can span dozens of lines), duplicates the per-symbol signature, and splits overloads across multiple rows.

Root cause: `generate_typedoc_module_index` rendered each category entry via `render_overview_line`, which emits `- [`name`](href) `kind` `signature` - summary`, the shared bullet helper used by the flat/category strategies.

## Summary

The TypeDoc module index now renders each category as a compact Markdown table; all changes are in `markdown.rs` (per-symbol pages, JSON, the HTML render style, and NAPI types are unchanged).

- Each category emits `## <Category>` followed by a `| <Singular> | Description |` table (`Function`, `Class`, `Interface`, `Type Alias`, `Enumeration`, `Variable`).
- One row per symbol, `[name](href)` linking to the per-symbol page; **the signature is no longer inlined** (it stays on the symbol page).
- **Overloads collapse to a single row** (same name → one entry).
- The description column is the symbol's first-paragraph summary, collapsed to one line, with `{@link}`/`{@linkcode}` resolved through the same renderer the per-symbol pages use (so cells show `An object that contains [argument schema](…).`, not raw `{@link}`), and table pipes escaped.
- `render_overview_line` itself is unchanged (still used by the flat/category strategies); the `## References` section (cross-entry-point re-exports) is left as-is.

## Risk

- Risk level: low / medium / high
- User or production impact:
- Rollback plan:

## Validation

- [ ] Automated tests or checks run:
- [ ] Manual verification completed:
- [ ] Edge cases or failure modes considered:

## Release and docs checklist

- [ ] Documentation, comments, or runbooks updated, or not needed.
- [ ] Configuration, migration, or environment changes documented, or not needed.
- [ ] Observability, logging, and alerting impact considered, or not needed.
- [ ] Security, privacy, and dependency impact considered, or not needed.

---
<a href="https://app.blacksmith.sh/ubugeeei-prod/codesmith/ox-content/pr/281"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1782874524&installation_id=136613492&pr_number=281&repository=ubugeeei-prod%2Fox-content&return_to=https%3A%2F%2Fgithub.com%2Fubugeeei-prod%2Fox-content%2Fpull%2F281&signature=03122c83edd2065fffe0a2d330aa3010b85d1143ae5b55b70f67390c9ad985d0"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by ubugeeei** [comment]
please fix ci

File: crates/ox_content_docs/src/markdown.rs [hunk 1]
```diff
@@ -374,6 +374,21 @@ fn typedoc_kind_title(kind: &str) -> &'static str {
...
+    match kind {
+        "function" => "Function",
+        "class" => "Class",
+        "interface" => "Interface",
+        "type" => "Type Alias",
+        "enum" => "Enumeration",
+        "variable" | "const" => "Variable",
+        "module" => "Module",
+        _ => "Symbol",
+    }
```
Comment (ubugeeei) [inline crates/ox_content_docs/src/markdown.rs]: maybe, we can use phf compile-time map

**Review by ubugeeei — CHANGES_REQUESTED** [review]

**Review by ubugeeei — APPROVED** [review]
